src/main/scala/collections/Steps.scala

```diff
@@ -0,0 +1,12 @@
+package collections
+
+object Steps extends App {
+  val N = 10
+  println(countSteps(N).size)
+
+  def countSteps(n: Int): LazyList[Int] = {
```

Comment:
you defined a function that produces a separate lazy list for each call, which is not efficient, rework it please to have a single lazy list for all calls

Reply:
changed the function implementation